gapic/src/main/com/google/gapid/perfetto/models/CounterTrack.java

```diff
@@ -137,15 +141,17 @@ public static Data empty(DataRequest req) {
     }
   }
 
-  public static class Values implements Selection, Selection.CombiningBuilder.Combinable<Values> {
+  public static class Values implements Selection<Long>, Selection.CombiningBuilder.Combinable<Values> {
     public final long[] ts;
     public final String[] names;
     public final double[][] values;
+    private final Set<Long> valueKeys = Sets.newHashSet();
 
     public Values(String name, Data data) {
       this.ts = data.ts;
       this.names = new String[] { name };
       this.values = new double[][] { data.values };
+      this.valueKeys.addAll(LongStream.of(ts).boxed().collect(Collectors.toList()));
```

Comment:
`LongStream.of(ts).boxed().forEach(this.valueKeys::add)` - avoids creating and collecting to the throw-away list.